src/box/sql/sqlInt.h

```diff
@@ -834,9 +834,9 @@ typedef u64 uptr;
  * pointers.  In that case, only verify 4-byte alignment.
  */
 #ifdef SQL_4_BYTE_ALIGNED_MALLOC
-#define EIGHT_BYTE_ALIGNMENT(X)   ((((char*)(X) - (char*)0)&3)==0)
+#define EIGHT_BYTE_ALIGNMENT(X)   ((((uintptr_t)((char *)(X))) & 3) == 0)
```

Comment:
We have way to set `SQL_4_BYTE_ALIGNED_MALLOC`
```Diff
diff --git a/src/box/sql/sqlInt.h b/src/box/sql/sqlInt.h
index 8e83743a6..07497f18d 100644
--- a/src/box/sql/sqlInt.h
+++ b/src/box/sql/sqlInt.h
@@ -828,16 +828,8 @@ typedef u64 uptr;
  * Assert that the pointer X is aligned to an 8-byte boundary.  This
  * macro is used only within assert() to verify that the code gets
  * all alignment restrictions correct.
- *
- * Except, if sql_4_BYTE_ALIGNED_MALLOC is defined, then the
- * underlying malloc() implementation might return us 4-byte aligned
- * pointers.  In that case, only verify 4-byte alignment.
  */
-#ifdef SQL_4_BYTE_ALIGNED_MALLOC
-#define EIGHT_BYTE_ALIGNMENT(X)   ((((uintptr_t)((char *)(X))) & 3) == 0)
-#else
 #define EIGHT_BYTE_ALIGNMENT(X)   ((((uintptr_t)((char *)(X))) & 7) == 0)
-#endif
 
 /*
  * Default maximum size of memory used by memory-mapped I/O in the VFS
```

Reply:
Done